src/Uno.UI/UI/Xaml/FrameworkElement.cs

```diff
@@ -442,16 +447,42 @@ private void OnRequestedThemeChanged(ElementTheme oldValue, ElementTheme newValu
 			{
 				// This is an ultra-naive implementation... but nonetheless enables the common use case of overriding the system theme for
 				// the entire visual tree (since Application.RequestedTheme cannot be set after launch)
+				// This will also explicitly change the Application.Current.RequestedTheme, which does not happen in case of UWP.
 				Application.Current.SetExplicitRequestedTheme(Uno.UI.Extensions.ElementThemeExtensions.ToApplicationThemeOrDefault(newValue));
 			}
+
+			var actualThemeChanged =
+				// 1. Previously was default, and new explicit value differs from application theme
+				(oldValue == ElementTheme.Default && Application.Current?.ActualElementTheme != newValue) ||
```

Comment:
Won't `ActualElementTheme` have already changed, since you call this after `SetExplicitRequestedTheme()`?

Reply:
That's expected - in this case I need to compare if previously the requested theme for this element was default (so it matched `Application.ActualElementTheme`), but now `RequestedTheme` is either Dark or Light, and I want to check if that new one is different from application's theme (in which case `ActualTheme` of this element changed)